### PR TITLE
Migrate provision-with-micromamba to setup-micromamba

### PR DESCRIPTION
**Pull request by weiji14**
**Description of proposed changes**

The mamba-org/provision-with-micromamba GitHub Action is about to be deprecated in favour of https://github.com/mamba-org/setup-micromamba, see https://github.com/mamba-org/provision-with-micromamba/pull/122. Also replaced the extra-specs parameter with create-args instead.

Migration reference: https://github.com/mamba-org/provision-with-micromamba/issues/114

Supersedes #2435

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by pavelzw** [comment]
You need to switch from 
```yml
        include:
          - python-version: '3.9'
            numpy-version: '1.21'
            optional-packages: ''
          - python-version: '3.11'
            numpy-version: '1.24'
            optional-packages: |
              contextily
              geopandas
              ipython
              rioxarray
              sphinx-gallery
[...]
            packaging
            ${{ matrix.optional-packages }}
            build
```

to

```yml
        include:
          - python-version: '3.9'
            numpy-version: '1.21'
            optional-packages: ''
          - python-version: '3.11'
            numpy-version: '1.24'
            optional-packages: >-
               contextily
              geopandas
              ipython
              rioxarray
              sphinx-gallery
[...]
            packaging${{ matrix.optional-packages }}
            build
```

(beware the extra space at `contextily`) since the new action expects all dependencies to be separated by spaces. 
The `|` YAML block scalar doesn't convert new lines to spaces but leaves them as newlines.

**Comment by weiji14** [comment]
> In [ci_tests.yaml](https://github.com/GenericMappingTools/pygmt/blob/setup-micromamba/.github/workflows/ci_tests.yaml#L70), you need to switch 
> to
> 
> ```yaml
>         include:
>           - python-version: '3.9'
>             numpy-version: '1.21'
>             optional-packages: ''
>           - python-version: '3.11'
>             numpy-version: '1.24'
>             optional-packages: >-
>                contextily
>               geopandas
>               ipython
>               rioxarray
>               sphinx-gallery
> [...]
>             packaging${{ matrix.optional-packages }}
>             build
> ```
> 
> (beware the extra space at `contextily`) since the new action expects all dependencies to be separated by spaces. The `|` YAML block scalar doesn't convert new lines to spaces but leaves them as newlines.

Thanks for the tip @pavelzw! I was wondering why the error for the CI test with optional dependencies was different to the one without. The single space before `contextily` didn't work at 58c807cbd49fae21c7eb1e06ba61adb5cdacc6a1, something about a YAML lint error. I just changed everything to a single line instead of multi-line at b5cc1c7831d60e1e0e7f7a7dd4764552b2544bce and now it runs!

File: .github/workflows/cache_data.yaml [hunk 1]
```diff
@@ -33,13 +33,16 @@ jobs:
           # fetch all history so that setuptools-scm works
           fetch-depth: 0
 
+      # Install Micromamba with conda-forge dependencies
       - name: Setup Micromamba
-        uses: mamba-org/provision-with-micromamba@v15
+        uses: mamba-org/setup-micromamba@v1.2.1
```
Comment (seisman) [inline .github/workflows/cache_data.yaml:38]: A new version was released
```suggestion
        uses: mamba-org/setup-micromamba@v1.3.0
```

Comment (weiji14) [reply .github/workflows/cache_data.yaml]: Done in 06c89e4367f09417949c49c0f4f9486931ad5af2

**Comment by seisman** [comment]
In PR https://github.com/GenericMappingTools/pygmt/pull/2435, I removed the `mamba list` step because the old `mamba-org/provision-with-micromamba` action already did it (https://github.com/GenericMappingTools/pygmt/pull/2435/files#r1173833125). 

I don't find the package list in the new action output. I think we should add `micromamba list` back.

**Review by seisman — APPROVED** [review]
Looks good.

**Comment by weiji14** [comment]
Hmm, the readthedocs build is failing with an unrelated `fiona.errors.DriverError: '/vsizip/vsicurl/` error. Might be due to changes with the geopandas 0.13.0 release (see https://github.com/geopandas/geopandas/pull/2796#discussion_r1191760864).

**Comment by pavelzw** [comment]
> I don't find the package list in the new action output. I think we should add `micromamba list` back.

I wanted to add this in `setup-micromamba` but forgot until now, thanks for reminding me :)

**Comment by pavelzw** [comment]
If you update to `v1.4.0` (or use `@v1` to always get the most recent `vx.y.z`), you can remove the `micromamba list` again.

**Pull request merged**